firstvoices/backend/serializers/page_serializers.py

```diff
@@ -38,3 +47,62 @@ class Meta(SiteContentLinkedTitleSerializer.Meta):
             "banner_image",
             "banner_video",
         )
+
+
+class SitePageDetailWriteSerializer(
+    UpdateSerializerMixin, CreateSiteContentSerializerMixin, SitePageDetailSerializer
+):
+    widgets = serializers.PrimaryKeyRelatedField(
+        queryset=SiteWidget.objects.all(),
+        many=True,
+    )
+    banner_image = serializers.PrimaryKeyRelatedField(
+        queryset=Image.objects.all(),
+        allow_null=True,
+        validators=[SameSite()],
+    )
+    banner_video = serializers.PrimaryKeyRelatedField(
+        queryset=Video.objects.all(),
+        allow_null=True,
+        validators=[SameSite()],
+    )
+
+    def validate_widgets(self, widgets):
+        if not widgets:
+            raise serializers.ValidationError(
+                "The widgets field must contain at least one widget ID."
+            )
+
+        site = get_site_from_context(self)
+        for site_widget in widgets:
+            if site_widget.site != site:
+                raise serializers.ValidationError(
+                    f"SiteWidget with ID ({site_widget.id}) does not belong to the site."
+                )
+        return widgets
+
+    def to_representation(self, instance):
+        data = SitePageDetailSerializer(instance=instance, context=self.context).data
+        return data
+
+    def create(self, validated_data):
+        validated_data["visibility"] = Visibility[
```

Comment:
Should we pull visibility handling into a mixin for all the controlled site content?

Reply:
I have added a mixin to do this. Thanks.